packages/cursorless-engine/src/processTargets/modifiers/getPreferredScopeTouchingPosition.ts

```diff
@@ -44,28 +38,13 @@ export function getPreferredScopeTouchingPosition(
    *  by one scope
    * - 2 scopes if position is between two adjacent scopes
    */
-  const candidates: TargetScope[] = [];
-
-  // FIXME: This will be a bit inefficient for paired delimiter scope handler,
-  // because we could end up walking all the way to the end of the file even if
-  // we're directly adjacent to a delimiter.  We could fix this by adding a
-  // `maxAncestorIndex` option to `ScopeIteratorRequirements` and using that in
-  // `ScopeHandler.generateScopes` for surrounding pair scope handler to stop
-  // early.
-  for (const scope of iterable) {
-    if (
-      !candidates.some((candidate) => scope.domain.contains(candidate.domain))
-    ) {
-      // We're only looking for minimal scopes
-      candidates.push(scope);
-    }
-
-    if (strictlyContains(scope.domain, position)) {
-      // If we've found a scope that strictly contains position,
-      // then we're not going to find any more minimal scopes
-      break;
-    }
-  }
+  const candidates: TargetScope[] = Array.from(
+    scopeHandler.generateScopes(editor, position, "forward", {
+      containment: "required",
+      allowAdjacentScopes: true,
+      maxAncestorIndex: 0,
```

Comment:
All of the logic we had here is now performed by `BaseScopeHandler` if `maxAncestorIndex` is 0